### PR TITLE
{network} Fix capture limit, size and time validation issue in `az network watcher packet-capture create`.

### PR DESCRIPTION
**Pull request by supriyajha-ms**
**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
network

**Review by wangzelin007 — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/network/_validators.py [hunk 1]
```diff
@@ -1869,6 +1869,18 @@ def validate_status_code_ranges(namespace):
...
+        if namespace.capture_limit < 0:
+            raise CLIError('usage error: --capture-limit cannot be a negative value.')
+    
+    if namespace.capture_size:
+        if namespace.capture_size < 0:
+            raise CLIError('usage error: --capture-size cannot be a negative value.')
+
+    if namespace.time_limit:
+        if namespace.time_limit < 0:
+            raise CLIError('usage error: --time-limit cannot be a negative value.')
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/_validators.py:1883]:
```suggestion

def validate_capture_size_and_limit(namespace):
    if namespace.capture_limit:
        if namespace.capture_limit < 0:
            raise CLIError('usage error: --capture-limit cannot be a negative value.')
    
    if namespace.capture_size:
        if namespace.capture_size < 0:
            raise CLIError('usage error: --capture-size cannot be a negative value.')

    if namespace.time_limit:
        if namespace.time_limit < 0:
            raise CLIError('usage error: --time-limit cannot be a negative value.')

```

**Review by kairu-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/network/_validators.py [hunk 2]
```diff
@@ -1870,6 +1870,20 @@ def validate_status_code_ranges(namespace):
     namespace.status_code_ranges = values
 
 
+def validate_capture_size_and_limit(namespace):
+    if namespace.capture_limit:
```
Comment (dhharkaw) [inline src/azure-cli/azure/cli/command_modules/network/_validators.py:1874]: What was the current state due to which the limit and size functionality was broken ?

File: src/azure-cli/azure/cli/command_modules/network/_validators.py [hunk 3]
```diff
@@ -1870,6 +1870,20 @@ def validate_status_code_ranges(namespace):
...
+            raise CLIError('usage error: --capture-limit cannot be a negative value.')
+
+    if namespace.capture_size:
+        if namespace.capture_size < 0:
+            raise CLIError('usage error: --capture-size cannot be a negative value.')
+
+    if namespace.time_limit:
+        if namespace.time_limit < 0:
+            raise CLIError('usage error: --time-limit cannot be a negative value.')
+
```
Comment (dhharkaw) [inline src/azure-cli/azure/cli/command_modules/network/_validators.py:1885]: Can we please add some unit tests for this ?

File: src/azure-cli/azure/cli/command_modules/network/_validators.py [hunk 4]
```diff
@@ -1870,6 +1870,20 @@ def validate_status_code_ranges(namespace):
...
+    if namespace.capture_limit:
+        if namespace.capture_limit < 0:
+            raise CLIError('usage error: --capture-limit cannot be a negative value.')
+
+    if namespace.capture_size:
+        if namespace.capture_size < 0:
+            raise CLIError('usage error: --capture-size cannot be a negative value.')
+
+    if namespace.time_limit:
+        if namespace.time_limit < 0:
```
[the remark this answers is not in the document]
Comment (supriyajha-ms) [reply src/azure-cli/azure/cli/command_modules/network/_validators.py:1883]: There is no higher limit. I have validated for int and non negative values.

File: src/azure-cli/azure/cli/command_modules/network/_validators.py [hunk 5]
```diff
@@ -1870,6 +1870,20 @@ def validate_status_code_ranges(namespace):
     namespace.status_code_ranges = values
 
 
+def validate_capture_size_and_limit(namespace):
+    if namespace.capture_limit:
+        if namespace.capture_limit < 0:
+            raise CLIError('usage error: --capture-limit cannot be a negative value.')
+
+    if namespace.capture_size:
+        if namespace.capture_size < 0:
```
[the remark this answers is not in the document]
Comment (supriyajha-ms) [reply src/azure-cli/azure/cli/command_modules/network/_validators.py:1879]: Yes, have done the testing and validation.

File: src/azure-cli/azure/cli/command_modules/network/_validators.py (hunk 2 above)
Comment (supriyajha-ms) [reply src/azure-cli/azure/cli/command_modules/network/_validators.py:1874]: There was no check for int type. It was also accepting strings and other values.

File: src/azure-cli/azure/cli/command_modules/network/_validators.py (hunk 3 above)
Comment (supriyajha-ms) [reply src/azure-cli/azure/cli/command_modules/network/_validators.py:1885]: Here the validation is only for negative values. I can add the UT in the next iteration with the change of support of vmss.

**Pull request merged**